src/test/test.java/com.teragrep.jos_01.procfs/ProcessTest.java

```diff
@@ -0,0 +1,116 @@
+ *
+ * Names of the licensors and authors may not be used for publicity purposes.
+ *
+ * No rights are granted for use of trade names, trademarks, or service marks
+ * which are in The Program if any.
+ *
+ * Licensee must indemnify licensors and authors for any liability that these
+ * contractual assumptions impose on licensors and authors.
+ *
+ * To the extent this program is licensed as part of the Commercial versions of
+ * Teragrep, the applicable Commercial License may apply to this file if you as
+ * a licensee so wish it.
+ */
+package com.teragrep.jos_01.procfs;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.util.ArrayList;
+import java.util.Map;
+
+public class ProcessTest {
+
+    @Test
+    public void readFileTest() {
+        Process systemd = new Process(1);
+        Map<String, String> systemdStats = systemd.stat().statistics();
+        Assert.assertEquals("1", systemdStats.get("pid"));
+        Assert.assertEquals("(systemd)", systemdStats.get("comm"));
+        Assert.assertEquals(52, systemdStats.size());
+        for (Map.Entry<String, String> entry : systemdStats.entrySet()) {
+            Assert.assertNotNull(entry.getValue());
+            Assert.assertNotNull(entry.getKey());
+        }
+
+        Process kthreadd = new Process(2);
+        Map<String, String> kthreaddStats = kthreadd.stat().statistics();
+        Assert.assertEquals("2", kthreaddStats.get("pid"));
+        Assert.assertEquals("(kthreadd)", kthreaddStats.get("comm"));
+        Assert.assertEquals(52, kthreaddStats.size());
+        for (Map.Entry<String, String> entry : kthreaddStats.entrySet()) {
+            Assert.assertNotNull(entry.getValue());
+            Assert.assertNotNull(entry.getKey());
+        }
+    }
+
+    @Test
+    public void constructorTest() {
+        Process process = new Process(1);
+        Process processString = new Process("1");
+    }
+
+    @Test
+    public void noSuchProcessTest() {
+        Process process = new Process(-1);
+        Assert.assertFalse(process.isAlive());
+    }
+
+    @Test
+    public void procFileNamesTest() {
+        Process process = new Process(1);
+        ArrayList<String> fileNames = process.availableProcFiles();
+        Assert.assertEquals(243, fileNames.size());
+    }
+
+    @Test
+    public void correctNumberOfTasksTest() {
+        Process process = new Process(1);
+        String statTaskcount = process.stat().statistics().get("num_threads");
+        int taskCount = process.tasks().size();
+        Assert.assertEquals(statTaskcount, String.valueOf(taskCount));
+    }
+
+    @Test
+    public void procTest() {
+        Process process = new Process(1);
+        ArrayList<String> rows = process.proc("status").rows();
+        Assert.assertEquals(59, rows.size());
+
+        rows = process.proc("comm").rows();
+        Assert.assertEquals(1, rows.size());
+        Assert.assertEquals("systemd", rows.get(0));
+    }
```

Comment:
Maybe we could consider adding test to see procDirectory and procFile give us expected instances?

Reply:
I found and fixed a bug in Process' constructor thanks to this! Process's procDirectory field was being hardcoded to "/proc" instead of using the same as LinuxOS object. I also added a test case for this bug.